## Задача 1

Оформляйте ноутбук, используя эти советы:

Номер задачи - ## заголовок2

Номер подзадачи - ### заголовок 3

Предоставленные наборы данных оформляйте, как код

## Задача 2
На складе лежат разные фрукты в разном количестве. Нужно написать функцию, которая на вход принимает любое количество названий фруктов и их количество, а возвращает общее количество фруктов на складе

In [29]:
def func(data: dict):
    sum = 0
    for i in data:
        sum += data[i]
    return sum
total_fruits = {
    'Айва': 301,
    'Груша': 5006,
    'Яблоко': 145,
    'Питахайя': 600,
    'Авокадо': 120
}


func(total_fruits)

6172

На складе всего 6172 фруктов.

## Задача 3

Дан список с затратами на рекламу. Но в данных есть ошибки, некоторые затраты имеют отрицательную величину. Удалите такие значения из списка и посчитайте суммарные затраты

In [30]:
expenses = [100, 125, -90, 345, 655, -1, 0, 200]

In [31]:
# в условии суммы добавлен ноль, так как это нулевые затраты, но на результат это не влияет
sum ([i for i in expenses if i >= 0])

1425

Затраты на рекламу равны 1425.

## Задача 4
Даны два списка.
Дата покупки:

In [32]:
data = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']

In [33]:
value = [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

### Задача 4.1
Найдите, какая выручка у компании в ноябре Используйте list comprehensions

In [34]:
sum ([i[1] for i in list(zip(data, value)) if "2021-11-" in i[0]])

25098

В ноябре компания выручила 25098

### Задача 4.2
Найдите выручку компании в зависимости от месяца Для этого напишите функцию, которая на вход принимает список с датами и список с выручкой, а на выходе словарь, где ключи - это месяцы, а значения - это выручка. Используйте аннотирование типов.

In [35]:
def profit(data1: list, data2: list) -> dict:
    db = [(i[0][5:7], i[1]) for i in zip(data1, data2)]
    dikt = {i[0]:0 for i in db}
    for i in db:
        dikt[i[0]] += i[1]
    return dikt

profit(data, value)
        
    

{'09': 25647, '12': 45452, '10': 28645, '11': 25098}